src/pytest_run_parallel/plugin.py

```diff
@@ -72,8 +72,17 @@ def inner(*args, **kwargs):
         failed = None
         barrier = threading.Barrier(n_workers)
         original_switch = sys.getswitchinterval()
+        new_switch = 1e-6
```

Comment:
```suggestion
        new_switch = 1e-9
```

The PR description says nanoseconds

Reply:
Oh yeah, sorry. Meant microseconds instead. The plan was to start with the original value and increase by an order of magnitude if it fails.

Reply:
Just out of curiosity - when can this code get triggered? Or I guess said another way, when can `sys.setswitchinterval` fail? I've never had any issues setting the switch interval to a small number.

Reply:
My understanding is that on i386 and i586 (and especially the latter), there has always been some weird behavior around floats. This can lead to a rounding error, where the C implementation of `sys.getswitchinterval` effectively sees a 0 and raises a `ValueError`.